core/src/test/java/org/bitcoinj/wallet/DefaultRiskAnalysisTest.java

```diff
@@ -253,6 +253,14 @@ public void transactionVersions() throws Exception {
 
         tx.setVersion(3);
         analysis = DefaultRiskAnalysis.FACTORY.create(wallet, tx, NO_DEPS);
+        assertEquals(RiskAnalysis.Result.OK, analysis.analyze());
+
+        tx.setType(Transaction.Type.TRANSACTION_NORMAL);
+        analysis = DefaultRiskAnalysis.FACTORY.create(wallet, tx, NO_DEPS);
+        assertEquals(RiskAnalysis.Result.OK, analysis.analyze());
```

Comment:
Not sure if we need a separate `version=3; type=normal` test?

Reply:
normal is the default.  

Though one thing we probably didn't do was put in a Rule in DefaultRiskAnalysis to check the value of type, that it is less than a particular value.  

I suppose we don't need to make a change based on type in this Story.  But, the List of types are here:
https://github.com/dashpay/dips/blob/master/dip-0002-special-transactions.md.  If we did add a check for valid transaction version 3 types, then the type would need to be less than 7.  Though 7 is proposed and won't become official for a few months probably.